File: data/derived_cot/rq1_traces/translation/r1/java_to_py/code_output/PersonRequest.py
class PersonRequest:
    def __init__(self, name, sex, phone_number):
        self._name = self._validate_name(name)
        self._sex = self._validate_sex(sex)
        self._phone_number = self._validate_phone_number(phone_number)
    
    def _validate_name(self, name):
        if name is None or name == '' or len(name) > 33:
            return None
        return name
    
    def _validate_sex(self, sex):
        if sex is None:
            return None
        if sex not in ['Man', 'Woman', 'UGM']:
            return None
        return sex
    
    def _validate_phone_number(self, phone_number):
        if phone_number is None or phone_number == '':
            return None
        if len(phone_number) != 11:
            return None
        for c in phone_number:
            if c not in '0123456789':
                return None
        return phone_number
    
    def getName(self):
        return self._name
    
    def getSex(self):
        return self._sex
    
    def getPhoneNumber(self):
        return self._phone_number

File: data/derived_cot/rq2_eval/translation/qwen/_converted_code/py_to_cpp/PersonRequest.cpp
#include <string>

class PersonRequest {
private:
    static bool is_all_digits(const std::string& sex) {
        return sex == "Man" || sex == "Woman" || sex == "UGM";
    }

    static bool is_valid_phone_number(const std::string& phone_number) {
        if (phone_number.empty()) return false;
        if (phone_number.length() != 11) return false;
        for (char c : phone_number) {
            if (!isdigit(static_cast<unsigned char>(c))) 
                return false;
        }
        return true;
    }

public:
    PersonRequest(const std::string& name, const std::string& sex, const std::string& phone_number) 
        : name(_validate_name(name)), 
          sex(_validate_sex(sex)), 
          phone_number(_validate_phone_number(phone_number)) {}

    const std::string& get_name() const { return name; }
    const std::string& get_sex() const { return sex; }
    const std::string& get_phone_number() const { return phone_number; }

private:
    std::string validate_name(const std::string& name) {
        if (name.empty() || name.length() > 33) 
            return "";
        return name;
    }

    std::string validate_sex(const std::string& sex) {
        if (!is_valid_sex(sex)) 
            return "";
        return sex;
    }

    std::string validate_phone_number(const std::string& phone_number) {
        if (!is_valid_phone_number(phone_number)) 
            return "";
        return phone_number;
    }

    std::string name;
    std::string sex;
    std::string phone_number;
};